### DISCUSSION TITLE
Is it okay to use constant sizes in PDF?

### DISCUSSION THREAD

**Discussion by aidrecabrera** [QuestPDF/QuestPDF#695] [Q&A] [answered]
Hey there!

I'm currently exploring the use of constant sizes and column sizes in QuestPDF. I've been accustomed to prioritizing responsiveness in web design, typically avoiding the use of constant sizes like pixels (px) for this purpose. Could anyone advise me on whether it's considered a good practice to use constant sizes and column sizes in QuestPDF? I know this is a dumb question, but I am just curious if it is okay and it does not matter?

Looking forward to your insights and suggestions!

Thank you!

**Comment by MarcinZiabek** [comment, marked as answer]
Great question!

Let's look over why responsiveness is so important in web design. Web pages are being displayed on all sorts of displays: of various sizes (phone, tablet, desktop), of various aspect ratios (square, 16:9, ultrawide), in a resizable browser window, with different DPI and scaling, with additional accessibility settings enabled, etc. The possibilities are enormous, and the only meaningful way to keep the page properly displaying is to follow responsive design guidelines.

On the other hand, paper is just paper. Its size is very well-defined. The A4 page is exactly 8.3 x 11.7 inches, while 1 inch = 72 points. That's it. Using constant dimensions is entirely ok. I personally believe that layouts defined with constant values may prove to be more scalable and more straightforward to debug and maintain. However, whenever calculating exact sizes is not practical (e.g. when you want to divide available horizontal space into 3 parts of equal width), feel free to use relative items.

That's my point of view. If anybody has a different perspective, please don't hesitate to share your point of view!

**Reply by aidrecabrera** [reply]
Now I understand why responsiveness is crucial in web design, unlike in PDFs, papers, and the like. I had been considering the implications of responsiveness, given my experience in web design and such. I had developed a strong inclination toward responsiveness (I got too dogmatic), and that led me to ask the question. Your clear and informative response has cleared up my doubts.

Another thing, I was wondering if you have a guide or resources that you can recommend on best practices for designing PDFs and similar documents, just as there are guides for web design responsiveness? It would be great to learn more about optimizing designs for this specific context.

Thanks, @MarcinZiabek! I really love your PDF library, it helped me a lot so much, especially as a beginner.

**Reply by MarcinZiabek** [reply]
As far as I see, designing PDFs (or documents in general) is an art of its own. It becomes more difficult whenever the content is changing (e.g.: amount of text, number of images), which is valid for most PDFs generated by software. There are surely some articles about this topic, please look for hints regarding DTP, UX, etc.

At some point, I would like to create a couple of layouts that could act as starting points. But it is lower in terms of priority.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
